src/components/plots/GeneSearchBar.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react';
+import _ from 'lodash';
+import { AutoComplete } from 'antd';
+import { useSelector, useDispatch } from 'react-redux';
+import { loadGeneExpression } from 'redux/actions/genes';
+import PropTypes from 'prop-types';
+
+const filterGenes = (searchText, geneList, loadedGenes) => {
+  const searchTextUpper = searchText.toUpperCase();
+  const filteredList = geneList.filter((gene) => gene.toUpperCase().includes(searchTextUpper));
+  const disableLoaded = filteredList.map((gene) => loadedGenes.includes(gene));
+
+  // options needs to be an array of objects
+  return filteredList.map((geneName, index) => ({ value: geneName, disabled: disableLoaded[index] }));
+};
+
+const GeneSearchBar = (props) => {
+  const { plotUuid, experimentId, searchBarUuid } = props;
+
+  const dispatch = useDispatch();
+
+  const geneList = useSelector((state) => state.genes.properties.views[searchBarUuid]?.data);
+
```

Comment:
These can also be stored under `state.genes.properties.data`. Why are you storing them under `state.genes.properties.views[searchBarUuid]?.data` instead?

Reply:
makes sense, thanks for the clarification